apps/server/src/modules/files-storage/service/files-storage.service.ts

```diff
@@ -1,47 +1,106 @@
-import { Injectable, InternalServerErrorException } from '@nestjs/common';
-import { Counted, FileRecord } from '@shared/domain';
+import { Injectable, InternalServerErrorException, NotAcceptableException, NotFoundException } from '@nestjs/common';
+import { Counted, EntityId, FileRecord, ScanStatus } from '@shared/domain';
+import { AntivirusService } from '@shared/infra/antivirus/antivirus.service';
 import { FileRecordRepo } from '@shared/repo';
 import { Logger } from '@src/core/logger';
 import { S3ClientAdapter } from '../client/s3-client.adapter';
-import { FileRecordParams, RenameFileParams, ScanResultParams, SingleFileParams } from '../controller/dto';
-import { FilesStorageHelper } from '../helper';
+import {
+	CopyFileResponse,
+	CopyFilesOfParentParams,
+	DownloadFileParams,
+	FileRecordParams,
+	RenameFileParams,
+	ScanResultParams,
+	SingleFileParams,
+} from '../controller/dto';
+import { ErrorType } from '../error';
+import {
+	createICopyFiles,
+	createPath,
+	deriveStatusFromSource,
+	getNewFileRecord,
+	getPaths,
+	getStatusFromScanResult,
+	isStatusBlocked,
+	mapFileRecordToFileRecordParams,
+	markForDelete,
+	modifyFileNameInScope,
+	resolveFileNameDuplicates,
+	unmarkForDelete,
+} from '../helper';
+import { IFile, IGetFileResponse } from '../interface';
 
 @Injectable()
 export class FilesStorageService {
 	constructor(
 		private readonly fileRecordRepo: FileRecordRepo,
 		private readonly storageClient: S3ClientAdapter,
-		private logger: Logger,
-		private readonly filesStorageHelper: FilesStorageHelper
+		private readonly antivirusService: AntivirusService,
+		private logger: Logger
 	) {
 		this.logger.setContext(FilesStorageService.name);
 	}
 
 	// find
-	public async getFile(params: SingleFileParams): Promise<FileRecord> {
+	public async getFileRecord(params: SingleFileParams): Promise<FileRecord> {
 		const fileRecord = await this.fileRecordRepo.findOneById(params.fileRecordId);
 
 		return fileRecord;
 	}
 
-	public async getFileMarkedForDelete(params: SingleFileParams) {
+	public async getFileRecordBySecurityCheckRequestToken(token: string): Promise<FileRecord> {
+		const fileRecord = await this.fileRecordRepo.findBySecurityCheckRequestToken(token);
+
+		return fileRecord;
+	}
+
+	public async getFileRecordMarkedForDelete(params: SingleFileParams) {
 		const fileRecord = await this.fileRecordRepo.findOneByIdMarkedForDelete(params.fileRecordId);
 
 		return fileRecord;
 	}
 
-	public async getFilesOfParent(params: FileRecordParams): Promise<Counted<FileRecord[]>> {
+	public async getFileRecordsOfParent(params: FileRecordParams): Promise<Counted<FileRecord[]>> {
 		const countedFileRecords = await this.fileRecordRepo.findBySchoolIdAndParentId(params.schoolId, params.parentId);
 
 		return countedFileRecords;
 	}
 
+	// upload
+	public async tryToCreateFileInStorage(
```

Comment:
try finde ich zu unspezifisch wir können ruhig sagen das wir im Fehler zurück rollen.

Reply:
Ja spezifisch finde ich gut. Komme nur nicht auf einen treffenden Namen, der nicht super lang ist. 🤔

Reply:
Wird in BC-2553 behandelt

Reply:
Mach es super lang. Bei den anderen haben wir irgendwas mit bla bla Rollback.

Reply:
Zu tryToCreateFileInStorageAndRollbackOnError umbenannt.